### ISSUE TITLE
Provider produced inconsistent result after apply when creating event stream topic 

### ISSUE THREAD

**Issue by sflaherty2009** [IBM-Cloud/terraform-provider-ibm#2275]
Hi there,

Thank you for opening an issue. Please note that we try to keep the Terraform issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.

### Terraform Version
Terraform v0.13.6

### Affected Resource(s)
Please list the resources as a list, for example:
-ibm_event_streams_topic

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration Files
```hcl
resource "ibm_event_streams_topic" "clima_matched_conditions" {
  resource_instance_id = ibm_resource_instance.es_instance.id
  name                 = "clima-matched-conditions"
  partitions           = 10
  config = {
    "cleanup.policy"  = "compact,delete"
    "retention.ms"    = "86400000"
    "retention.bytes" = "1073741824"
    "segment.bytes"   = "536870912"
  }
}
```

### Debug Output
Please provider a link to a GitHub Gist containing the complete debug output: https://www.terraform.io/docs/internals/debugging.html. Please do NOT paste the debug output in the issue; just paste a link to the Gist.

### Panic Output
If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the 
```
Error: Provider produced inconsistent result after apply

When applying changes to ibm_event_streams_topic.clima_matched_conditions,
provider "registry.terraform.io/ibm-cloud/ibm" produced an unexpected new
value: Root resource was present, but now absent.

This is a bug in the provider, which should be reported in the provider's own
issue tracker.
```

### Expected Behavior
Created a list of topics for the IBM Event Stream

### Actual Behavior
Only created some of the topics. 

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

**Comment by hkantare** [comment]
We are bot able to reproduce it can you let us know the stpes you followed 
```
ibm_resource_instance.es_instance: Creation complete after 21s [id=crn:v1:bluemix:public:messagehub:us-south:a/883079c85357a1f3f85d968780e56518:164bfab0-9284-4ee4-bb5d-cc4c1fba4180::]
ibm_event_streams_topic.es_topic: Creating...
2021/02/24 18:05:58 [DEBUG] ibm_event_streams_topic.es_topic: applying the planned Create change
2021-02-24T18:05:58.988+0530 [INFO]  plugin.terraform-provider-ibm: 2021/02/24 18:05:58 [DEBUG] setting computed for "kafka_brokers_sasl" from ComputedKeys: timestamp=2021-02-24T18:05:58.988+0530
2021-02-24T18:05:59.171+0530 [INFO]  plugin.terraform-provider-ibm: 2021/02/24 18:05:59 [INFO] createSaramaAdminClient kafka_http_url is set to https://7jtbnlqtchlc6kzq.svc05.us-south.eventstreams.cloud.ibm.com: timestamp=2021-02-24T18:05:59.171+0530
2021-02-24T18:05:59.171+0530 [INFO]  plugin.terraform-provider-ibm: 2021/02/24 18:05:59 [INFO] createSaramaAdminClient kafka_brokers_sasl is set to [broker-1-7jtbnlqtchlc6kzq.kafka.svc05.us-south.eventstreams.cloud.ibm.com:9093 broker-0-7jtbnlqtchlc6kzq.kafka.svc05.us-south.eventstreams.cloud.ibm.com:9093 broker-3-7jtbnlqtchlc6kzq.kafka.svc05.us-south.eventstreams.cloud.ibm.com:9093 broker-2-7jtbnlqtchlc6kzq.kafka.svc05.us-south.eventstreams.cloud.ibm.com:9093 broker-5-7jtbnlqtchlc6kzq.kafka.svc05.us-south.eventstreams.cloud.ibm.com:9093 broker-4-7jtbnlqtchlc6kzq.kafka.svc05.us-south.eventstreams.cloud.ibm.com:9093]: timestamp=2021-02-24T18:05:59.171+0530
2021-02-24T18:06:03.529+0530 [INFO]  plugin.terraform-provider-ibm: 2021/02/24 18:06:03 [INFO] createSaramaAdminClient instance crn:v1:bluemix:public:messagehub:us-south:a/883079c85357a1f3f85d968780e56518:164bfab0-9284-4ee4-bb5d-cc4c1fba4180:: 's client is initialized: timestamp=2021-02-24T18:06:03.528+0530
2021-02-24T18:06:07.680+0530 [INFO]  plugin.terraform-provider-ibm: 2021/02/24 18:06:07 [INFO] resourceIBMEventStreamsTopicCreate CreateTopic: topic is myeventtopic, detail is {10 3 map[] map[cleanup.policy:0xc00042af50 retention.bytes:0xc00042af80 retention.ms:0xc00042af70 segment.bytes:0xc00042af40]}: timestamp=2021-02-24T18:06:07.680+0530
2021-02-24T18:06:07.896+0530 [INFO]  plugin.terraform-provider-ibm: 2021/02/24 18:06:07 [INFO] createSaramaAdminClient kafka_http_url is set to https://7jtbnlqtchlc6kzq.svc05.us-south.eventstreams.cloud.ibm.com: timestamp=2021-02-24T18:06:07.896+0530
2021-02-24T18:06:07.896+0530 [INFO]  plugin.terraform-provider-ibm: 2021/02/24 18:06:07 [INFO] createSaramaAdminClient kafka_brokers_sasl is set to [broker-1-7jtbnlqtchlc6kzq.kafka.svc05.us-south.eventstreams.cloud.ibm.com:9093 broker-0-7jtbnlqtchlc6kzq.kafka.svc05.us-south.eventstreams.cloud.ibm.com:9093 broker-3-7jtbnlqtchlc6kzq.kafka.svc05.us-south.eventstreams.cloud.ibm.com:9093 broker-2-7jtbnlqtchlc6kzq.kafka.svc05.us-south.eventstreams.cloud.ibm.com:9093 broker-5-7jtbnlqtchlc6kzq.kafka.svc05.us-south.eventstreams.cloud.ibm.com:9093 broker-4-7jtbnlqtchlc6kzq.kafka.svc05.us-south.eventstreams.cloud.ibm.com:9093]: timestamp=2021-02-24T18:06:07.896+0530
ibm_event_streams_topic.es_topic: Still creating... [10s elapsed]
2021-02-24T18:06:10.324+0530 [INFO]  plugin.terraform-provider-ibm: 2021/02/24 18:06:10 [INFO] createSaramaAdminClient instance crn:v1:bluemix:public:messagehub:us-south:a/883079c85357a1f3f85d968780e56518:164bfab0-9284-4ee4-bb5d-cc4c1fba4180:: 's client is initialized: timestamp=2021-02-24T18:06:10.324+0530
ibm_event_streams_topic.es_topic: Creation complete after 13s [id=crn:v1:bluemix:public:messagehub:us-south:a/883079c85357a1f3f85d968780e56518:164bfab0-9284-4ee4-bb5d-cc4c1fba4180:topic:myeventtopic]
2021-02-24T18:06:12.035+0530 [WARN]  plugin.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = transport is closing"
2021-02-24T18:06:12.039+0530 [DEBUG] plugin: plugin process exited: path=/Users/harinireddy/terraform/terraform-provider-ibm pid=64542
2021-02-24T18:06:12.039+0530 [DEBUG] plugin: plugin exited

Apply complete! Resources: 2 added, 0 changed, 0 destroyed.
```
```
data "ibm_resource_group" "group" {
  name = "default"
}
resource "ibm_resource_instance" "es_instance" {
  name              = "myevent"
  service           = "messagehub"
  plan              = "standard"
  location          = "us-south"
  resource_group_id = data.ibm_resource_group.group.id
}
resource "ibm_event_streams_topic" "es_topic" {
  resource_instance_id = ibm_resource_instance.es_instance.id
  name                 = "myeventtopic"
  partitions           = 10
  config = {
    "cleanup.policy"  = "compact,delete"
    "retention.bytes" = "86400000"
    "retention.ms"    = "1073741824"
    "segment.bytes"   = "536870912"
  }
}

```

**Comment by sflaherty2009** [comment]
This appears to be an issue with the number of topic partitions. Evidently there is a hard limit at 100 partitions. I have been hitting that and this is the error that I see when that max number is reached. Not sure if there is a way to spit out a better error or not.

**Comment by hkantare** [comment]
Ok..May be I will forwarded or ask this ticket to look by Event stream service...

**Comment by hkantare** [comment]
@JunliWang Can you please look into this issue...
@sflaherty2009 Can you please share the sample terraform configuration file

**Comment by hkantare** [comment]
@sflaherty2009 Can you please share complete Terraform plan /apply log after you set export TF_LOG=debug mode

**Comment by kavya498** [comment]
Available in 1.37.0 release..
Closing this issue..
Thanks..
